src/test/kotlin/com/terraformation/backend/funder/FundingEntityServiceTest.kt

```diff
@@ -0,0 +1,263 @@
+import com.terraformation.backend.funder.db.FundingEntityNotFoundException
+import com.terraformation.backend.funder.db.FundingEntityStore
+import com.terraformation.backend.funder.db.FundingEntityUserStore
+import com.terraformation.backend.mockUser
+import io.mockk.every
+import java.time.Instant
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.security.access.AccessDeniedException
+
+class FundingEntityServiceTest : DatabaseTest(), RunsAsUser {
+  override val user: TerrawareUser = mockUser()
+
+  private val clock = TestClock()
+  private lateinit var fundingEntityStore: FundingEntityStore
+  private lateinit var fundingEntityUserStore: FundingEntityUserStore
+  private lateinit var fundingEntityId: FundingEntityId
+  private lateinit var service: FundingEntityService
+
+  @BeforeEach
+  fun setUp() {
+    fundingEntityStore = FundingEntityStore(dslContext)
+    fundingEntityUserStore = FundingEntityUserStore(dslContext)
+    service = FundingEntityService(clock, dslContext, fundingEntityStore, fundingEntityUserStore)
+
+    fundingEntityId = insertFundingEntity()
+    insertOrganization()
+
+    every { user.canReadFundingEntities() } returns true
+    every { user.canCreateFundingEntities() } returns true
+    every { user.canDeleteFundingEntities() } returns true
+    every { user.canUpdateFundingEntities() } returns true
+    every { user.canUpdateFundingEntityProjects() } returns true
+  }
+
+  @Test
+  fun `create requires user to be able to manage funding entities`() {
+    every { user.canCreateFundingEntities() } returns false
+
+    assertThrows<AccessDeniedException> { service.create("Some Other Entity") }
+  }
+
+  @Test
+  fun `create populates created and modified fields`() {
+    val newTime = clock.instant().plusSeconds(1000)
+    clock.instant = newTime
+
+    val newUserId = insertUser()
+
+    val name = "New Funding Entity"
+    val row = FundingEntitiesRow(name = name)
+
+    every { user.userId } returns newUserId
+
+    val createdModel = service.create(row.name!!)
+
+    val expected =
+        row.copy(
+            createdBy = user.userId,
+            createdTime = clock.instant(),
+            id = createdModel.id,
+            modifiedBy = newUserId,
+            modifiedTime = clock.instant(),
+            name = name,
+        )
+    // retrieving both row and model to check createdBy/modifiedBy as well as projects
+    val actualRow = fundingEntitiesDao.fetchOneById(createdModel.id)!!
+    val actualModel = fundingEntityStore.fetchOneById(createdModel.id)
+
+    assertEquals(expected, actualRow)
+    assertEquals(0, actualModel.projects!!.size)
+  }
+
+  @Test
+  fun `create rejects duplicates by name`() {
+    assertThrows<FundingEntityExistsException> { service.create("TestFundingEntity") }
```

Comment:
To be concurrency-safe, this will either need to fetch the funding entity that's inserted in `setUp` (presuming that `insertFundingEntity` is changed to use randomized names) or use its own randomized name and try to create it twice so it can verify the second creation fails.